src/signal.rs

```diff
@@ -469,6 +483,17 @@ impl GnssSignal {
     }
 }
 
+#[cfg(feature = "sbp-conversions")]
+impl std::convert::TryFrom<sbp::messages::gnss::GnssSignal> for GnssSignal {
+    type Error = InvalidGnssSignal;
+
+    fn try_from(value: sbp::messages::gnss::GnssSignal) -> Result<GnssSignal, InvalidGnssSignal> {
+        use std::convert::TryInto;
+
+        GnssSignal::new(value.sat as u16, value.code.try_into()?)
+    }
+}
```

Comment:
@notoriaga Think I should add conversions from borrowed instances as well? All of these conversions don't contain large structures that can be efficiently moved, so we don't gain anything by only consuming the input value.

Reply:
Maybe it would be possible to use `AsRef` so we don't need to duplicate the conversion functions?

Reply:
Looking at the stdlib for some guidance, it seems like they only implement `From` and `TryFrom` for non-borrowed types except for types that can only by borrowed (i.e. `str` and slices). So it doesn't seem like the intended use for us to borrow with these converters.